File: app/src/test/java/info/nightscout/androidaps/plugins/general/automation/elements/LabelWithElementTest.java
package info.nightscout.androidaps.plugins.general.automation.elements;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.powermock.modules.junit4.PowerMockRunner;

@RunWith(PowerMockRunner.class)
public class LabelWithElementTest {

    @Test
    public void constructorTest() {
        LabelWithElement l = new LabelWithElement("A", "B", new InputInsulin());
        Assert.assertEquals("A", l.textPre);
        Assert.assertEquals("B", l.textPost);
        Assert.assertEquals(InputInsulin.class, l.element.getClass());
    }

}

File: automation/src/test/java/info/nightscout/androidaps/plugins/general/automation/elements/LabelWithElementTest.kt
package info.nightscout.androidaps.plugins.general.automation.elements

import info.nightscout.androidaps.plugins.general.automation.triggers.TriggerTestBase
import org.junit.Assert
import org.junit.Test

class LabelWithElementTest : TriggerTestBase() {

    @Test
    fun constructorTest() {
        val l = LabelWithElement(rh, "A", "B", InputInsulin())
        Assert.assertEquals("A", l.textPre)
        Assert.assertEquals("B", l.textPost)
        Assert.assertEquals(InputInsulin::class.java, l.element!!.javaClass)
    }
}